sdks/python/apache_beam/io/restriction_trackers.py

```diff
@@ -171,7 +175,7 @@ def checkpoint(self):
       else:
         end_position = self._current_position + 1
 
-      residual_range = (end_position, self._range.stop)
+      residual_range = OffsetRange(end_position, self._range.stop)
```

Comment:
In which case this could be

`self._range, residual_range = self._range.split(end_position)`